implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -213,10 +240,11 @@ public void onComplete() {
             } else {
                 final Publisher<String> publisher;
                 if (statsdConfig.buffered()) {
-                    publisher = BufferingFlux.create(this.sink.asFlux(), "\n", statsdConfig.maxPacketLength(), statsdConfig.pollingFrequency().toMillis())
+                    publisher = BufferingFlux.create(this.sink.asFlux().publish().autoConnect(), "\n",
+                                                     statsdConfig.maxPacketLength(), statsdConfig.pollingFrequency().toMillis())
                             .onBackpressureLatest();
                 } else {
-                    publisher = this.sink.asFlux();
+                    publisher = this.sink.asFlux().publish().autoConnect();
```

Comment:
actually -  this may be optional. I don't know if we have tests for that but if we do I will try to remove that and put only in cases where we have retry In place